File: Models/Company.cs
namespace LedgerFlow.API.Models
{
    /// <summary>
    /// A tenant in the multi-tenant system.
    /// </summary>
    public class Company
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string? Phone { get; set; }
        public string? Address { get; set; }
        public string? ContactPerson { get; set; }

        public int PlanId { get; set; }
        public Plan? Plan { get; set; }

        /// <summary>Quarterly or Yearly</summary>
        public string BillingCycle { get; set; } = "Quarterly";

        /// <summary>Trial | Active | Expired | Suspended</summary>
        public string SubscriptionStatus { get; set; } = "Trial";

        /// <summary>Maximum number of users allowed for this company</summary>
        public int MaxUsers { get; set; } = 10;

        /// <summary>When the trial period ends (if applicable)</summary>
        public DateTime? TrialEndsAt { get; set; }

        /// <summary>When the subscription started</summary>
        public DateTime? SubscriptionStartedAt { get; set; }

        /// <summary>When the subscription expires (null = no expiry)</summary>
        public DateTime? SubscriptionExpiresAt { get; set; }

        /// <summary>Active | Suspended (legacy field, kept for compatibility)</summary>
        public string Status { get; set; } = "Active";

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: ledgerflow-frontend/src/types/company.ts
/**
 * Company Types
 * Standardized company interfaces for the entire application
 */

export type SubscriptionStatus = 'Trial' | 'Active' | 'Expired' | 'Suspended';
export type CompanyStatus = 'Active' | 'Suspended' | 'Inactive';

export interface Plan {
  id: number;
  name: string;
  description: string;
  quarterlyPrice: number;
  yearlyPrice: number;
  maxUsers: number;
  maxExpensesPerMonth: number;
  canUploadReceipt: boolean;
  hasAdvancedReports: boolean;
  hasAdvancedAnalytics: boolean;
  hasDepartmentAnalytics: boolean;
  hasRoleBasedWorkflows: boolean;  // ✅ CHANGED: Renamed from hasMultiLevelApprovals
  hasPrioritySupport: boolean;
  trialDays: number;
}

export interface Company {
  id: number;
  name: string;
  email: string;
  phone?: string;
  address?: string;
  contactPerson?: string;
  planId: number;
  planName: string;
  billingCycle: string;
  subscriptionStatus: SubscriptionStatus;
  status: CompanyStatus;
  maxUsers?: number;
  trialEndsAt?: string;
  subscriptionStartedAt?: string;
  subscriptionExpiresAt?: string;
  userCount: number;
  expenseCount?: number;
  createdAt: string;
  updatedAt?: string;
}

export interface CompanyDetail extends Company {
  expensesThisMonth: number;
}

export interface CompanyListItem {
  id: number;
  name: string;
  email: string;
  planName: string;
  subscriptionStatus: SubscriptionStatus;
  trialEndsAt?: string;
  subscriptionExpiresAt?: string;
  userCount: number;
  expenseCount?: number;
  createdAt: string;
}

export interface CreateCompanyRequest {
  companyName: string;
  companyEmail: string;
  companyPhone?: string;
  companyAddress?: string;
  adminFirstName: string;
  adminLastName: string;
  adminEmail: string;
  adminPassword: string;
  planId: number;
}

export interface UpdateCompanyRequest {
  companyId: number;
  name?: string;
  email?: string;
  phone?: string;
  address?: string;
  planId?: number;
}

export interface ExtendTrialRequest {
  companyId: number;
  days: number;
}

export interface UpdatePlanRequest {
  companyId: number;
  planId: number;
}

export interface UpdateSubscriptionStatusRequest {
  companyId: number;
  subscriptionStatus: SubscriptionStatus;
  subscriptionExpiresAt?: string;
}
